activerecord/test/cases/connection_management_test.rb

```diff
@@ -62,12 +62,11 @@ def test_connections_closed_if_exception
       end
 
       def test_connections_not_closed_if_exception_and_test
-        executor.wrap do
```

Comment:
`wrap` here was hiding the bug. as checking the `active_connections?` was happening before the `clear_` method was running.